### DISCUSSION TITLE
How to I  distribute mivus across cluster? What is the simplest way to do on premise? How do I set specific nodes, so after restart all data is here.

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#21985] [Q&A and General discussion]
How to I  distribute mivus across cluster? What is the simplest way to do on premise? How do I set **specific nodes,** so after restart all data is here.

Can someone share his own experisence? Is rancher an adequate tool for the job to a simple cluster?
How do I configure a milvus cluster?

**Comment by yhmo** [comment]
It is recommended to use helm to deploy milvus. You need to spend some time to get familiar with the deployment tools.

**Reply by martinenkoEduard** [reply]
Can someone share his own experience? 
I mean how to change amount  nodes...

**Reply by xiaofan-luan** [reply]
see document here https://milvus.io/docs/scaleout.md

**Reply by martinenkoEduard** [reply]
How do I ensure data consistency?

**Reply by yhmo** [reply]
> How do I ensure data consistency?

https://milvus.io/docs/v2.1.x/consistency.md
Milvus supports four consistency levels: strong, bounded staleness, session, and eventually.
In search() interface, you can use the parameter "consistency_level" to specify consistency level:
```
esults = collection.search(
	data=[[0.1, 0.2]], 
	anns_field="book_intro", 
	param=search_params, 
	limit=10, 
	consistency_level="Strong"
)
```

"Strong" level is slow, it waits hundreds of milliseconds to ensure all the data is consumed.
"Bounded" level allows data inconsistency during a certain period of time. 
"Session" level ensure all the data that inserted by a client session is consumed
"Eventually" level ignore data consistency, execute the search request immediatly.

**Reply by martinenkoEduard** [reply]
No I mean that the nodes start at the same physical machines.

**Reply by martinenkoEduard** [reply]
Does it use Persistent Volume by default?
And does it use a stateful set as a default configuration?

**Reply by yhmo** [reply]
All the nodes use the same storage service: minio or s3.
No matter the nodes are deployed on the same machine or different machines. The storage has only one place.

**Reply by martinenkoEduard** [reply]
but on the load the data is distributed across different nodes? 
And it loads from minio? 

So it is possible to use external minio server to keep data on hard disk and then when cluster starts it loads data into RAM in chunks

**Reply by LoveEachDay** [reply]
Milvus relies on the default storageclass to provision pv automatically for object storage, etcd storage, and message queue storage.
